homeassistant/components/watchyourlan/sensor.py

```diff
@@ -0,0 +1,134 @@
+"""Support for the WatchYourLAN service."""
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .coordinator import WatchYourLANUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the WatchYourLAN sensors."""
+    coordinator = WatchYourLANUpdateCoordinator(hass, entry.data)
+    await coordinator.async_config_entry_first_refresh()
+
+    entities = [WatchYourLANDeviceCountSensor(coordinator)]
+
+    if isinstance(coordinator.data, list):
+        entities += [
+            WatchYourLANSensor(coordinator, device, entry.data)
+            for device in coordinator.data
+            if isinstance(device, dict)
+        ]
+
+    async_add_entities(entities)
+
+
+class WatchYourLANSensor(SensorEntity):
+    """Representation of a WatchYourLAN sensor."""
+
+    def __init__(
+        self, coordinator: WatchYourLANUpdateCoordinator, device: dict, config: dict
+    ) -> None:
+        """Initialize the sensor for a single device."""
+        super().__init__()
+        self.coordinator = coordinator
+        self.device = device
+        self.base_url = config["url"]
+
+    @property
+    def name(self):
+        """Return the name of the sensor, using the format 'WatchYourLAN <ID> <Optionally include name>'."""
+        device_id = self.device.get("ID")
+        device_name = self.device.get("Name", "").strip()
+
+        if not device_name:
+            return f"WatchYourLAN {device_id}"
+        return f"WatchYourLAN {device_id} {device_name}"
+
+    @property
+    def unique_id(self):
+        """Return a unique ID for this sensor based solely on the device's ID."""
+        return f"watchyourlan_{self.device.get('ID')}"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return "Online" if self.device.get("Now") == 1 else "Offline"
+
+    @property
+    def extra_state_attributes(self):
+        """Return other attributes related to the device, including the link to the device's page."""
+        host_id = self.device.get("ID")
+        host_page_url = f"{self.base_url}/host/{host_id}"
+
+        return {
+            "ID": self.device.get("ID"),
+            "Name": self.device.get("Name"),
+            "IP": self.device.get("IP"),
+            "Mac": self.device.get("Mac"),
+            "Hardware": self.device.get("Hw"),
+            "Iface": self.device.get("Iface"),
+            "DNS": self.device.get("DNS"),
+            "Last_Seen": self.device.get("Date"),
+            "Known": self.device.get("Known"),
+            "Now": self.device.get("Now"),
+            "Host Page": host_page_url,
+        }
```

Comment:
A lot of these datapoints can be their own sensor

Reply:
Is it preferred to have them be their own sensors, or attributes of a single sensor? I can certainly make them individual sensors if that is the preferred approach, but that seemed messy because it created so many sensors.

Reply:
We want entities rather than attributes. They're more efficient for the database and more customizable for the user. Also, some of these fields should be stored via the device info